### ISSUE TITLE
[Development] Add pyproject.toml

### ISSUE THREAD

**Issue by martinvonk** [pastas/pastas#442]
We need to add an pyproject.toml

Useful reference: https://ianhopkinson.org.uk/2022/02/understanding-setup-py-setup-cfg-and-pyproject-toml-in-python/

Warning I encountered:
DEPRECATION: ... is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559

I believe @dbrakenhoff already encountered some issues with this in [@pastastore](https://github.com/pastas/pastastore/commit/08d9f3017fb313ff5989604586f5e43130b1f1e2)

**Comment by raoulcollenteur** [comment]
OK, not a fan of moving everything into a src/pastas folder but if that is what Pypa wants I think this should be done.

Let make a 0.22.0 release first, and right after that merge the TypeHinting PR and move the whole thing into src folder.

I kinda feel like we should release a 0.23.0b to master to check if the GH workflow for packaging still works...

**Comment by martinvonk** [comment]
Closing this since #443 is merged.

**Comment by martinvonk** [comment]
I foresee a problem here when creating the next release :)

https://github.com/pastas/pastas/blob/17c6fd4960d5adf7fa0bf645e03a98691c0dcb6f/.github/workflows/python-publish.yml#L30

**Comment by raoulcollenteur** [comment]
Yes, can you fix this? Also, do we still really need the version.py with the new format? And if so, we could move `show_version`, `check_numba` and `check_numbascipy` to version.py?
